corgi/tasks/common.py

```diff
@@ -79,6 +81,15 @@ def create_relations(build_ids, build_type, external_system_id, product_ref, rel
             defaults={"type": relation_type},
         )
         if created:
+            # Similar to fetch_unprocessed_relations
+            # This skips use of the Collector models for builds in the CENTOS koji instance
+            # It was done to avoid updating the collector models not to use build_id as
+            # a primary key. It's possible because the only product stream (openstack-rdo)
+            # stored in CENTOS koji doesn't use modules
```

Comment:
The problem with the current design is that that the fetch_unprocessed_relation task doesn't update the product taxonomy for a build if that build already exists. With the modifications in this PR we now update the product taxonomy for a build even if it already exists, whenever a relation is created.

I think we could probably remove the scheduled fetch_unprocessed_relations tasks, but I hesitated to do that straight away, as I wanted to monitor the task output to make sure they were always return 0 (indicated they didn't find any relations without build objects) because removing them.

We could change the design so that instead of passing a callback function to create_relations we always call slow_fetch_*_build even if a SoftwareBuild exists in order to make sure that builds product taxonomy is up to date. However I think the current design is increases cohesion by keeping creation of relations and they representation in builds together.

If we call a task which is already in the backlog the Celery Singleton mechanism kicks in and ensures we only have one instance of that task in the backlog, so it shouldn't cause a redis backlog overload.